nsv13/code/game/objects/items/bloody_eye.dm

```diff
@@ -0,0 +1,191 @@
+	name = "Bloody Eye"
+	description = "A powerful drug of Syndicate origin. Manufactured to give the user an incredibly exciting rush, typically exilerating the user into a state of rage-induced psychosis. Boosts physical strength and allows the user to ignore the effects of critical condition."
+	reagent_state = LIQUID
+	color = "#D92323"
+	overdose_threshold = 5
+	addiction_threshold = 5
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+
+/datum/reagent/drug/bloody_eye/on_mob_metabolize(mob/living/L)
+	L.apply_status_effect(/datum/status_effect/bloody_eye)
+	..()
+
+//You won't go down unless you get killed with excess damage.
+
+/datum/reagent/drug/bloody_eye/on_mob_end_metabolize(mob/living/L)
+	L.remove_status_effect(/datum/status_effect/bloody_eye)
+	..()
+
+/datum/reagent/drug/bloody_eye/on_mob_life(mob/living/carbon/M)
+	if(prob(5))
+		var/high_message = pick("You see red.", "You feel like ripping out someone's throat.", "You feel like nothing could ever kill you.")
+		to_chat(M, "<span class='notice'>[high_message]</span>")
+	M.AdjustStun(-40, FALSE)
+	M.AdjustKnockdown(-40, FALSE)
+	M.AdjustUnconscious(-40, FALSE)
+	M.AdjustParalyzed(-40, FALSE)
+	M.AdjustImmobilized(-40, FALSE)
+	M.adjustStaminaLoss(-30, 0)
+	M.Jitter(2)
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 0.25)
+	if(prob(5))
+		INVOKE_ASYNC(M, /mob.proc/emote, pick("scream", "laugh"))
+	..()
+	. = 1
+
+/datum/reagent/drug/bloody_eye/overdose_start(mob/living/M)
+	to_chat(M, "<span class='userdanger'>KILLKILLKILLKILLKILLKILLKILLKILL!</span>")
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "[type]_overdose", /datum/mood_event/overdose, name)
+
+/datum/reagent/drug/bloody_eye/overdose_process(mob/living/M)
+	if((M.mobility_flags & MOBILITY_MOVE) && !ismovableatom(M.loc))
+		for(var/i in 1 to 4)
+			step(M, pick(GLOB.cardinals))
+	if(prob(20))
+		INVOKE_ASYNC(M, /mob.proc/emote, pick("scream", "laugh"))
+	if(prob(33))
+		M.visible_message("<span class='danger'>[M]'s eyes are blood red!</span>")
+		M.drop_all_held_items()
+	..()
+	M.adjustToxLoss(0.25, 0)
+	M.adjustOrganLoss(ORGAN_SLOT_BRAIN, pick(0.1, 0.15, 0.2, 0.25))
+	. = 1
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage1(mob/living/M)
+	M.Jitter(20)
+	if(prob(20))
+		INVOKE_ASYNC(M, /mob.proc/emote, pick("scream", "laugh"))
+	..()
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage2(mob/living/M)
+	M.Jitter(50)
+	M.Dizzy(5)
+	if(prob(30))
+		INVOKE_ASYNC(M, /mob.proc/emote, pick("scream", "laugh"))
+	..()
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage3(mob/living/M)
+	if((M.mobility_flags & MOBILITY_MOVE) && !ismovableatom(M.loc))
+		for(var/i = 0, i < 4, i++)
+			step(M, pick(GLOB.cardinals))
+	M.Jitter(100)
+	M.Dizzy(10)
+	if(prob(40))
+		INVOKE_ASYNC(M, /mob.proc/emote, pick("scream", "gasp"))
+	..()
+
+/datum/reagent/drug/bloody_eye/addiction_act_stage4(mob/living/carbon/human/M)
+	if((M.mobility_flags & MOBILITY_MOVE) && !ismovableatom(M.loc))
+		for(var/i = 0, i < 8, i++)
+			step(M, pick(GLOB.cardinals))
```

Comment:
Also use Move(), step() is a byond proc and doesn't have the any checks besides density

Reply:
Can't find a good example of how to use the Move() proc, got a suggestion?

Reply:
Aye, it's an atom movable proc
```M.Move(get_step(get_turf(src), pick(GLOB.cardinals))```

Reply:
regarding the step speed, the only (good) solution I can think of is to cut down the number of steps. IMO just change the frequency of steps depending on OD level rather than the amount of steps, keeping them all at 1.

Reply:
> regarding the step speed, the only (good) solution I can think of is to cut down the number of steps. IMO just change the frequency of steps depending on OD level rather than the amount of steps, keeping them all at 1.

just sleep()?

Reply:
in a proc called on processing? 😆 
The sleeps for all of those steps probably wouldn't have even finished by the time the next call comes around

Reply:
Literally just slow down the processing then